src/redisai.c

```diff
@@ -721,6 +724,17 @@ struct RedisAI_RunInfo {
   RAI_Error* err;
 };
 
+struct RedisAI_RunStats {
+  RedisModuleString *key;
+  int type; // model or script
+  RAI_Backend backend;
+  char* devicestr;
+  long long duration_us;
```

Comment:
@lantiga based on total duration in us and total calls we can calculate the average model/script run time. My question goes if the average is enough right now or do we want extended metrics like quantiles?

Reply:
Quantiles would certainly be interesting, @filipecosta90 do you have a suggestion for computing running quantiles?

Reply:
(see e.g. https://sites.cs.ucsb.edu/~suri/psdir/ency.pdf)

Reply:
@lantiga we're working on the usage of histograms specifically target for latency measurements on other projects ( HDR histogram ). I will revise your link and we can discuss this afterwards on a meeting :) Independently of the code used, the whole idea of extended metrics is a must IMHO.